manta-js/package/src/pallets/MantaPayWallet.ts

```diff
@@ -0,0 +1,176 @@
+import BN from 'bn.js';
+import type {
+  Wallet as WasmWallet,
+  Transaction as WasmTransaction,
+} from '../wallet/crate/pkg/manta_wasm_wallet';
+import type {
+  Address,
+  IMantaPayWallet,
+  SignedTransaction,
+  IBaseWallet,
+  ILedgerApi,
+  PalletName,
+  Network,
+} from '.././interfaces';
+import {
+  mapPostToTransaction,
+  privateTransferBuildUnsigned,
+  toPrivateBuildUnsigned,
+  toPublicBuildUnsigned,
+  transactionsToBatches,
+  transferPost,
+  wrapWasmError,
+} from '../utils';
+import PrivateWallet from '../PrivateWallet';
+
+const CURRENT_PALLET_NAME: PalletName = 'mantaPay';
+
+/// PrivateWallet class
+export default class MantaPayWallet
+  extends PrivateWallet
+  implements IMantaPayWallet
+{
+  constructor(
+    network: Network,
+    baseWallet: IBaseWallet,
+    wasmWallet: WasmWallet,
+    ledgerApi: ILedgerApi,
+  ) {
+    super(CURRENT_PALLET_NAME, network, baseWallet, wasmWallet, ledgerApi);
+  }
+
+  static init(network: Network, baseWallet: IBaseWallet): MantaPayWallet {
+    const params = PrivateWallet.getInitialParams(
+      CURRENT_PALLET_NAME,
+      baseWallet,
+    );
+    return new MantaPayWallet(
+      network,
+      baseWallet,
+      params.wasmWallet,
+      params.ledgerApi,
+    );
+  }
+
+  /// Builds and signs a "To Private" transaction for any fungible token.
+  /// Note: This transaction is not published to the ledger.
+  async toPrivateBuild(
```

Comment:
So there are no more of the `send` variants, that were using polkadot-js api to submit the transactions on-chain ?

Reply:
Will not add that method, signAndSend is just one line of code, and can customize nonce

Reply:
Yeah I agree, this is cleaner.